e2etests/tests/import_vrf.go

```diff
@@ -0,0 +1,432 @@
+		}
+		err := updater.Clean()
+		Expect(err).NotTo(HaveOccurred())
+
+		cs = k8sclient.New()
+	})
+
+	initNeighbors := func(useVrf bool, ipFamily ipfamily.Family) ([]*frrcontainer.FRR, config.PeersConfig, []frrk8sv1beta1.Neighbor) {
+		frrs := config.ContainersForVRF(infra.FRRContainers, "")
+		if useVrf {
+			frrs = config.ContainersForVRF(infra.FRRContainers, infra.VRFName)
+		}
+		peersConfig := config.PeersForContainers(frrs, ipFamily)
+		neighbors := config.NeighborsFromPeers(peersConfig.PeersV4, peersConfig.PeersV6)
+		return frrs, peersConfig, neighbors
+	}
+
+	pairWithNodes := func(frrs []*frrcontainer.FRR, family ipfamily.Family, toAdvertise []string) {
+		for _, c := range frrs {
+			err := container.PairWithNodes(cs, c, family, func(frr *container.FRR) {
+				frr.NeighborConfig.ToAdvertiseV4 = address.FilterForFamily(toAdvertise, ipfamily.IPv4)
+				frr.NeighborConfig.ToAdvertiseV6 = address.FilterForFamily(toAdvertise, ipfamily.IPv6)
+			})
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+
+	updateAndCheckPeered := func(config frrk8sv1beta1.FRRConfiguration, peersDefault, peersVRF config.PeersConfig, frrsDefault, frrsVRF []*frrcontainer.FRR, family ipfamily.Family) {
+		err := updater.Update(peersDefault.Secrets, config)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = updater.Update(peersVRF.Secrets, config)
+		Expect(err).NotTo(HaveOccurred())
+
+		nodes, err := k8s.Nodes(cs)
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, c := range frrsDefault {
+			ValidateFRRPeeredWithNodes(nodes, c, family)
+		}
+		for _, c := range frrsVRF {
+			ValidateFRRPeeredWithNodes(nodes, c, family)
+		}
+	}
+
+	baseConfig := frrk8sv1beta1.FRRConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: k8s.FRRK8sNamespace,
+		},
+		Spec: frrk8sv1beta1.FRRConfigurationSpec{
+			BGP: frrk8sv1beta1.BGPConfig{
+				Routers: []frrk8sv1beta1.Router{
+					{
+						ASN: infra.FRRK8sASN,
+					},
+					{
+						ASN: infra.FRRK8sASNVRF,
+						VRF: infra.VRFName,
+					},
+				},
+			},
+		},
+	}
+
+	prefixesToSelectors := func(prefixes []string) []frrk8sv1beta1.PrefixSelector {
+		res := []frrk8sv1beta1.PrefixSelector{}
+		for _, p := range prefixes {
+			selector := frrk8sv1beta1.PrefixSelector{
+				Prefix: p,
+				LE:     32,
+				GE:     0,
+			}
+			ip, _, err := net.ParseCIDR(p)
+			if err != nil {
+				panic(err)
+			}
+			if ip.To4() == nil {
+				selector.LE = 64
+			}
+			res = append(res, selector)
+		}
+		return res
+	}
+
+	ginkgo.Context("while receiving IPs", func() {
+
+		ginkgo.DescribeTable("when the default VRF imports a vrf", func(
+			ipFamily ipfamily.Family,
+			toAdvertise,
+			toAdvertiseVRF []string,
+			allowMode frrk8sv1beta1.AllowMode) {
+
+			frrsDefault, peersDefault, neighborsDefault := initNeighbors(false, ipFamily)
+			frrsVRF, peersVRF, neighborsVRF := initNeighbors(true, ipFamily)
+
+			ginkgo.By("pairing with nodes")
+			pairWithNodes(frrsDefault, ipFamily, toAdvertise)
```

Comment:
I would prefer is all ginkgo asser takes place not nested == function to return error and assert here